### PR TITLE
[Security] evaluate route aliases when checking if a path matches a request

### LINKED ISSUE #53736: [Security] `check_path` does not work with new FQCN route alias
### Symfony version(s) affected

7.0.3

### Description

The [newly introduced FQCN route aliases](https://symfony.com/blog/new-in-symfony-6-4-fqcn-based-routes) aren't working for `check_path` of a form login.

### How to reproduce

With this setup in `config/packages/security.php` my login form isn't working (i.e. the page is refreshed, but the login process isn't happening):
```php
$mainFirewall->formLogin()->checkPath(FooController::class);
```
However, when I pass the traditional route name (`'app_foo__invoke'`) or path (`/foo`), it works as expected.

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #53736
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Security/Http/HttpUtils.php [hunk 1]
```diff
@@ -109,7 +111,19 @@ public function checkRequestPath(Request $request, string $path): bool
...
+                if ($path === $request->attributes->get('_route')) {
+                    return true;
+                }
+
+                if (null === $this->urlGenerator) {
+                    return false;
+                }
+
+                try {
+                    return $request->getPathInfo() === $this->urlGenerator->generate($path);
```
Comment (ThomasLandauer) [inline src/Symfony/Component/Security/Http/HttpUtils.php:123]: The left side gives me `/second`, but the right side gives me `/symfony-issue-53736/public/index.php/second` (in the reproducer at https://github.com/ThomasLandauer/symfony-issue-53736)

Comment (xabbuh) [reply src/Symfony/Component/Security/Http/HttpUtils.php:123]: What web server do you use and how do you start/configure it? Using the Symfony CLI I don't have the `/symfony-issue-53736/public/index.php` part on the right side.

Comment (ThomasLandauer) [reply src/Symfony/Component/Security/Http/HttpUtils.php:123]: To get the reproducer running quickly, I used my "quick-and-dirty" server which has `localhost` configured to the parent directory of `symfony-issue-53736`. So in the browser I have http://localhost/symfony-issue-53736/public/index.php
The docs probably advise against this, but it usually works, so I'm not sure if this is an edge case or not...

**Comment by xabbuh** [comment]
closing here as the PR apparently doesn’t (fully) fix the reported issue

**Pull request closed without merging**